WordPressKit/WordPressOrgXMLRPCApi.swift

```diff
@@ -451,3 +506,109 @@ extension WordPressOrgXMLRPCApiError: LocalizedError {
         }
     }
 }
+
+public struct WordPressOrgXMLRPCApiFault: LocalizedError, HTTPURLResponseProviding {
+    var response: HTTPAPIResponse<Data>
+
+    var code: Int?
+    var message: String?
+
+    public var errorDescription: String? {
+        message
+    }
+
+    var httpResponse: HTTPURLResponse? {
+        response.response
+    }
+}
+
+private extension WordPressAPIResult<HTTPAPIResponse<Data>, WordPressOrgXMLRPCApiFault> {
+
+    func decodeXMLRPCResult() -> WordPressAPIResult<HTTPAPIResponse<AnyObject>, WordPressOrgXMLRPCApiFault> {
+        // This is a re-implementation of `WordPressOrgXMLRPCApi.handleResponseWithData` function:
+        // https://github.com/wordpress-mobile/WordPressKit-iOS/blob/11.0.0/WordPressKit/WordPressOrgXMLRPCApi.swift#L265
+        flatMap { response in
+            guard let contentType = response.response.allHeaderFields["Content-Type"] as? String else {
+                return .failure(.unparsableResponse(response: response.response, body: response.body, underlyingError: WordPressOrgXMLRPCApiError.unknown))
+            }
+
+            if (400..<600).contains(response.response.statusCode) {
+                if let decoder = WPXMLRPCDecoder(data: response.body), decoder.isFault() {
+                    // when XML-RPC is disabled for authenticated calls (e.g. xmlrpc_enabled is false on WP.org),
+                    // it will return a valid fault payload with a non-200
+                    return .failure(.endpointError(.init(response: response, code: decoder.faultCode(), message: decoder.faultString())))
+                } else {
+                    return .failure(.unacceptableStatusCode(response: response.response, body: response.body))
+                }
+            }
+
+            if ["application/xml", "text/xml"].filter({ (type) -> Bool in return contentType.hasPrefix(type)}).count == 0 {
```

Comment:
```suggestion
            if ["application/xml", "text/xml"].filter({ (type) -> Bool in return contentType.hasPrefix(type)}).isEmpty {
```

From SwiftLint, I learned that `isEmpty` is more efficient than `count == 0`. I'm pretty sure the difference is not something we would notice at runtime, but I still think it's a good practice to use `isEmpty`.

Reply:
I have simplified this code in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/719/commits/aeb9becfefd00339e339232c8bfd5d3605f854c2